### ISSUE TITLE
Toolbox GDAL  Raster Projection Warp Batch Creation  

### ISSUE THREAD

**Issue by mfstanton** [qgis/QGIS#49514]
### What is the bug or the crash?

When using the TOOLBOX GDAL Raster Projection Warp Batch Creation  script the first thing I notice is that the first row of the task setup is populated (row 2) with the last Layer selected in the Layers Panel of the Desktop UI. When the "Fill Down" is activated with Select Open Layers it is populated with the open layers plus a second copy of whatever layer was in the First row of the setup window. Attempting to remove the original Row 2 from the task list does not work.   If not observed before completing the setup steps, this leaves two tasks to reproject the Layer where the second one fails giving an error message in the job log. While only one job failing is a minor rerun action on the part of the user. That first failure of the Reproject job leaves an error setting somewhere that prevents the remainder of the jobs in the list, which could be lengthy for some projects, to fail to reproject, writing the original TIFF to the output folder and continuing on for the remainder of the files.  I have a log file that contains the rejection of the second attempt to reproject the tile but it is rather sizeable
![Double_Selection_First_EntryScreenshot 2022-07-29 111909](https://user-images.githubusercontent.com/23061117/181809701-d65d2e34-2acf-4a74-89f1-0c5fee6871d3.png)
![Selected_Last_Layer_Screenshot 2022-07-29 105421](https://user-images.githubusercontent.com/23061117/181809703-71bad92d-9b72-4c33-a271-e18b6ba4be2d.png)
![Selected_Layer_Screenshot 2022-07-29 105421](https://user-images.githubusercontent.com/23061117/181809705-6a075f72-a652-4bdb-aeeb-461f7133de0a.png)
![Select_All_Screenshot 2022-07-28 200030](https://user-images.githubusercontent.com/23061117/181809797-2bf796e1-9fad-4264-a876-4e6b913a30be.png)

[Short_Reproject_Log_Document.txt](https://github.com/qgis/QGIS/files/9222138/Short_Reproject_Log_Document.txt)
 

### Steps to reproduce the issue

Open Layer with input images 
Open Toolbox and go to GDAL Raster Projection, Warp (reproject) 
Right click on menu item Select Batch processing option 
When spreadsheet pops up look at the contents  of Row 2 where you will see an entry that corresponds to the last selected Layer. This entry cannot be removed. If you right click on the Inputs field Autofill option you will get a menu which gives you an option to select the Open Layers. 
Click the entry and when the popup shows click "select all" then OK. You will see two entries for the layer that was in the initial window. Running your job will error because there will be an attempt to put a second copy of the file into the folder.

### Versions

Buenos  Aires  3.26.1 

Was doing the same on the prior versions 3.26.0

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

I am trying to reproject from a UTM CRS to a  Missouri State Plane West projection.    

The workaround is to unselect the item that shows up in the Row 2 of the Layer list when selectinf layers from the popup .
Note that the remainder of the output files after the one with the error will ALSO not  be reprojected and stay in the same CRS as that of the input layers.  There seems to be a persisted error flag that keeps the reproject from happening if this error occurs

**Comment by gioman** [comment]
> Open Layer with input images
> Open Toolbox and go to GDAL Raster Projection, Warp (reproject)
> Right click on menu item Select Batch processing option
> When spreadsheet pops up look at the contents of Row 2 where you will see an entry that corresponds to the last selected Layer. This entry cannot be removed. If you right click on the Inputs field Autofill option you will get a menu which gives you an option to select the Open Layers.
> Click the entry and when the popup shows click "select all" then OK. You will see two entries for the layer that was in the initial window. Running your job will error because there will be an attempt to put a second copy of the file into the folder.

@mfstanton really hard to read description, but let me see if I understand, the problem stands from the fact that you get a duplicate entry in batch mode once you choose to "Select from Open Layers" and you cannot remove the duplicate entry, that is correct? if yes then I cannot replicate on 3.26.1 on Windows 10.

**Comment by mfstanton** [comment]
Yes that is correct.  I get around it by opening batch mode and unselecting the one that was picked as the last referenced.  The problem is that  the whole process is sloppy and generates an error because there was one image highlighted at first and IT gets picked to be in the list twice.  Not sure what makes it so hard to read.  Once the selection was made the screenshot shows two entries for the same image and the extra cannot be deleted. So far I have been able to do 20,000 images by unchecking the first image in the list (select from open layers popup)  as it is the one being duplicated and the menu gets built correctly.

**Comment by gioman** [comment]
I edited the title for sake of clarity.

**Comment by spatialthoughts** [comment]
I can reproduce this on the current QGIS 3.22.16 LTR as well. This is a regression from earlier versions and is not a good user experience. When using Autofill from 'Select from Open Layers' , there is already 1 row with one of the layers and if that layer is also selected, it creates a duplicate entry. It should skip the layer that is already present in the first row.
